### ISSUE TITLE
Embedding btcd/btcwallet in an Electron desktop GUI

### ISSUE THREAD

**Issue by msafi** [btcsuite/btcd#1581]
I'm building a full-node/wallet desktop GUI with Electron. I don't wanna build the full-node or wallet back-ends, just the GUI. I'm interested in using `btcd` and `btcwallet` to service the GUI. Are portable executables available of these programs?

**Comment by torkelrogstad** [comment]
You'd have to cross compile btcd and btcwallet to the different architectures you want to support, but that should be as simple as setting `GOARCH` and `GOOS` environment variables. For an example of a project that embeds a Golang binary into an Electron app, check out [Lightning App](https://github.com/lightninglabs/lightning-app/) by Lightning Labs.

**Comment by jcvernaleo** [comment]
For decred, the original version of decrediton did this where it was an electron app that included dcrd and dcrwallet and ran them in the background so it is definitely possible.

Go binaries are pretty much self contained, just have to build for your arch (although electron has pretty limited arch/os option anyway so go and btcd work easily on everything electron works on and more).

**Comment by msafi** [comment]
I was able to migrate my app from Bitcoin Core to btcd. It was a relatively pain-free process! I ran into a few Bitcoin Core compatibility issues (like mempool info missing some data, `getBlock` response missing `nTx` field, and `getrawtransaction` expecting an `int` instead of a `boolean` for the `verbose` field) but other than these minor issues, it worked well.

Building the `btcd` binaries for macOS and Windows was really just a matter of setting environment variables like @torkelrogstad mentioned. That was 🤯

I made a release of the new app https://github.com/orange-org/orange/releases/tag/1.0.0-beta.6 I would appreciate feedback from anyone who could try it out. 

Although there's a lot of infrastructure work in the app, it's still not much more than a bare-bone block explorer.

I'm now debating what to work on next, whether to continue adding GUI support for the other btcd PRC commands, or if I should bring in btcwallet and start working on the wallet. I would really love to have SPV support before doing the wallet work. But I don't see many people wanting to use a desktop app just for btcd's functionality...that's why I feel I should get started on the wallet...any thoughts on this?

Anyway, I think this issue can now be closed. Thanks for your help folks!
